src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerConfiguration\FixerOptionValidatorGenerator;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Andreas Möller <am@localheinz.com>
+ * @author SpacePossum
+ */
+final class BlankLineBeforeStatementFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * @var array
+     */
+    private static $tokenMap = [
+        'break' => T_BREAK,
+        'continue' => T_CONTINUE,
+        'declare' => T_DECLARE,
+        'do' => T_DO,
+        'for' => T_FOR,
+        'foreach' => T_FOREACH,
+        'if' => T_IF,
+        'include' => T_INCLUDE,
+        'include_once' => T_INCLUDE_ONCE,
+        'require' => T_REQUIRE,
+        'require_once' => T_REQUIRE_ONCE,
+        'return' => T_RETURN,
+        'switch' => T_SWITCH,
+        'throw' => T_THROW,
+        'try' => T_TRY,
+        'while' => T_WHILE,
+    ];
+
+    /**
+     * @var array
+     */
+    private $fixTokenMap = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->fixTokenMap = [];
+
+        foreach ($this->configuration['statements'] as $key) {
+            $this->fixTokenMap[$key] = self::$tokenMap[$key];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'An empty line feed should precede a configured statement.',
```

Comment:
well thats an English language catch I think; as here there is a one-on-one relation which will be applied multiple times. I.e. 
>An empty line feed should precede configured statements.

to me suggests that the line would _only_ be added if there is more than one statement, which is not true.
So;

>An empty line feed should precede any configured statement.

?